XamarinCommunityToolkit/Views/GravatarImageSource/GravatarHandlerUtil.shared.cs

```diff
@@ -0,0 +1,100 @@
+#if !NETSTANDARD
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	static class GravatarHandlerUtil
+	{
+		const string requestUriFormat = "https://www.gravatar.com/avatar/{0}?s={1}&d={2}";
+		static readonly Lazy<HttpClient> lazyHttp = new Lazy<HttpClient>(() => new HttpClient());
+
+		public static async Task<FileInfo> Load(ImageSource imageSource, float scale, string cacheDirectory)
+		{
+			if (imageSource is GravatarImageSource gis)
+			{
+				var cacheFilePath = Path.Combine(cacheDirectory, nameof(GravatarImageSource), CacheFileName(gis, scale));
+				var cacheFileInfo = new FileInfo(cacheFilePath);
+
+				if (!UseCacheFile(gis.CachingEnabled, gis.CacheValidity, cacheFileInfo))
+				{
+					var imageBytes = await GetGravatarAsync(gis.Email, gis.Size, scale, gis.Default).ConfigureAwait(false);
+					SaveImage(cacheFileInfo, imageBytes);
+				}
+
+				return cacheFileInfo;
+			}
+
+			return null;
+		}
+
+		static void SaveImage(FileInfo cacheFileInfo, byte[] imageBytes)
+		{
+			if (imageBytes.Length < 1)
```

Comment:
Should we check imageBytes for null?

Reply:
it shouldn't ever be null since any 200 response will give you an image and any non 200 response gives an empty array